File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/backend/php/fenix/src/model/cadastros/Colaborador.php
<?php
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [COLABORADOR] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="COLABORADOR")
 */
class Colaborador extends ModelBase implements \JsonSerializable
{
	/**
	 * @ORM\Id
	 * @ORM\Column(type="integer")
	 * @ORM\GeneratedValue(strategy="AUTO")
	 */
	private $id;

	/**
	 * @ORM\Column(type="string", name="MATRICULA")
	 */
	private $matricula;

	/**
	 * @ORM\Column(type="date", name="DATA_CADASTRO")
	 */
	private $dataCadastro;

	/**
	 * @ORM\Column(type="date", name="DATA_ADMISSAO")
	 */
	private $dataAdmissao;

	/**
	 * @ORM\Column(type="date", name="DATA_DEMISSAO")
	 */
	private $dataDemissao;

	/**
	 * @ORM\Column(type="string", name="CTPS_NUMERO")
	 */
	private $ctpsNumero;

	/**
	 * @ORM\Column(type="string", name="CTPS_SERIE")
	 */
	private $ctpsSerie;

	/**
	 * @ORM\Column(type="date", name="CTPS_DATA_EXPEDICAO")
	 */
	private $ctpsDataExpedicao;

	/**
	 * @ORM\Column(type="string", name="CTPS_UF")
	 */
	private $ctpsUf;

	/**
	 * @ORM\Column(type="string", name="OBSERVACAO")
	 */
	private $observacao;


    /**
     * Relations
     */
    
    /**
     * @ORM\OneToOne(targetEntity="Usuario", mappedBy="colaborador", cascade={"persist", "remove"})
     */
    private $usuario;

    /**
     * @ORM\OneToOne(targetEntity="Pessoa", fetch="EAGER")
     * @ORM\JoinColumn(name="ID_PESSOA", referencedColumnName="id")
     */
    private $pessoa;

    /**
     * @ORM\OneToOne(targetEntity="Cargo", fetch="EAGER")
     * @ORM\JoinColumn(name="ID_CARGO", referencedColumnName="id")
     */
    private $cargo;

    /**
     * @ORM\OneToOne(targetEntity="Setor", fetch="EAGER")
     * @ORM\JoinColumn(name="ID_SETOR", referencedColumnName="id")
     */
    private $setor;


    /**
     * Gets e Sets
     */

    public function getId() 
	{
		return $this->id;
	}

	public function setId($id) 
	{
		$this->id = $id;
	}

    public function getMatricula() 
	{
		return $this->matricula;
	}

	public function setMatricula($matricula) 
	{
		$this->matricula = $matricula;
	}

    public function getDataCadastro() 
	{
		if ($this->dataCadastro != null) {
			return $this->dataCadastro->format('Y-m-d');
		} else {
			return null;
		}
	}
	public function setDataCadastro($dataCadastro) 
	{
		$this->dataCadastro = $dataCadastro != null ? new \DateTime($dataCadastro) : null;
	}

    public function getDataAdmissao() 
	{
		if ($this->dataAdmissao != null) {
			return $this->dataAdmissao->format('Y-m-d');
		} else {
			return null;
		}
	}
	public function setDataAdmissao($dataAdmissao) 
	{
		$this->dataAdmissao = $dataAdmissao != null ? new \DateTime($dataAdmissao) : null;
	}

    public function getDataDemissao() 
	{
		if ($this->dataDemissao != null) {
			return $this->dataDemissao->format('Y-m-d');
		} else {
			return null;
		}
	}
	public function setDataDemissao($dataDemissao) 
	{
		$this->dataDemissao = $dataDemissao != null ? new \DateTime($dataDemissao) : null;
	}

    public function getCtpsNumero() 
	{
		return $this->ctpsNumero;
	}

	public function setCtpsNumero($ctpsNumero) 
	{
		$this->ctpsNumero = $ctpsNumero;
	}

    public function getCtpsSerie() 
	{
		return $this->ctpsSerie;
	}

	public function setCtpsSerie($ctpsSerie) 
	{
		$this->ctpsSerie = $ctpsSerie;
	}

    public function getCtpsDataExpedicao() 
	{
		if ($this->ctpsDataExpedicao != null) {
			return $this->ctpsDataExpedicao->format('Y-m-d');
		} else {
			return null;
		}
	}
	public function setCtpsDataExpedicao($ctpsDataExpedicao) 
	{
		$this->ctpsDataExpedicao = $ctpsDataExpedicao != null ? new \DateTime($ctpsDataExpedicao) : null;
	}

    public function getCtpsUf() 
	{
		return $this->ctpsUf;
	}

	public function setCtpsUf($ctpsUf) 
	{
		$this->ctpsUf = $ctpsUf;
	}

    public function getObservacao() 
	{
		return $this->observacao;
	}

	public function setObservacao($observacao) 
	{
		$this->observacao = $observacao;
	}

    public function getUsuario(): ?Usuario 
	{
		return $this->usuario;
	}

	public function setUsuario(?Usuario $usuario) 
	{
		$this->usuario = $usuario;
	}

    public function getPessoa(): ?Pessoa 
	{
		return $this->pessoa;
	}

	public function setPessoa(?Pessoa $pessoa) 
	{
		$this->pessoa = $pessoa;
	}

    public function getCargo(): ?Cargo 
	{
		return $this->cargo;
	}

	public function setCargo(?Cargo $cargo) 
	{
		$this->cargo = $cargo;
	}

    public function getSetor(): ?Setor 
	{
		return $this->setor;
	}

	public function setSetor(?Setor $setor) 
	{
		$this->setor = $setor;
	}


    /**
     * Constructor
     */
    public function __construct($objetoJson)
    {
        if (isset($objetoJson)) {
            isset($objetoJson->id) ? $this->setId($objetoJson->id) : $this->setId(null);
            $this->mapear($objetoJson);
        }

		if (isset($objetoJson->usuario)) {
			$this->setUsuario(new Usuario($objetoJson->usuario));
			$this->getUsuario()->setColaborador($this);
		}

		
    }

    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
			$this->setMatricula($objeto->matricula);
			$this->setDataCadastro($objeto->dataCadastro);
			$this->setDataAdmissao($objeto->dataAdmissao);
			$this->setDataDemissao($objeto->dataDemissao);
			$this->setCtpsNumero($objeto->ctpsNumero);
			$this->setCtpsSerie($objeto->ctpsSerie);
			$this->setCtpsDataExpedicao($objeto->ctpsDataExpedicao);
			$this->setCtpsUf($objeto->ctpsUf);
			$this->setObservacao($objeto->observacao);
		}
    }


    /**
     * Validation
     */
    public function validarObjetoRequisicao($objJson, $id)
    {
        return parent::validarObjeto($objJson, $id, 'Colaborador');
    }


    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getId(),
			'matricula' => $this->getMatricula(),
			'dataCadastro' => $this->getDataCadastro(),
			'dataAdmissao' => $this->getDataAdmissao(),
			'dataDemissao' => $this->getDataDemissao(),
			'ctpsNumero' => $this->getCtpsNumero(),
			'ctpsSerie' => $this->getCtpsSerie(),
			'ctpsDataExpedicao' => $this->getCtpsDataExpedicao(),
			'ctpsUf' => $this->getCtpsUf(),
			'observacao' => $this->getObservacao(),
			'usuario' => $this->getUsuario(),
			'pessoa' => $this->getPessoa(),
			'cargo' => $this->getCargo(),
			'setor' => $this->getSetor(),
        ];
    }
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/frontend/provider/lib/src/model/cadastros/colaborador.dart
/*
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [COLABORADOR] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';

import 'package:intl/intl.dart';
import 'package:fenix/src/model/model.dart';

class Colaborador {
	int id;
	int idPessoa;
	int idCargo;
	int idSetor;
	String matricula;
	DateTime dataCadastro;
	DateTime dataAdmissao;
	DateTime dataDemissao;
	String ctpsNumero;
	String ctpsSerie;
	DateTime ctpsDataExpedicao;
	String ctpsUf;
	String observacao;
	Usuario usuario;
	Pessoa pessoa;
	Cargo cargo;
	Setor setor;

	Colaborador({
			this.id,
			this.idPessoa,
			this.idCargo,
			this.idSetor,
			this.matricula,
			this.dataCadastro,
			this.dataAdmissao,
			this.dataDemissao,
			this.ctpsNumero,
			this.ctpsSerie,
			this.ctpsDataExpedicao,
			this.ctpsUf,
			this.observacao,
			this.usuario,
			this.pessoa,
			this.cargo,
			this.setor,
		});

	static List<String> campos = <String>[
		'ID', 
		'MATRICULA', 
		'DATA_CADASTRO', 
		'DATA_ADMISSAO', 
		'DATA_DEMISSAO', 
		'CTPS_NUMERO', 
		'CTPS_SERIE', 
		'CTPS_DATA_EXPEDICAO', 
		'CTPS_UF', 
		'OBSERVACAO', 
	];
	
	static List<String> colunas = <String>[
		'Id', 
		'Matrícula', 
		'Data de Cadastro', 
		'Data de Admissão', 
		'Data de Demissão', 
		'Número CTPS', 
		'Série CTPS', 
		'Data de Expedição', 
		'UF CTPS', 
		'Observação', 
	];

	Colaborador.fromJson(Map<String, dynamic> jsonDados) {
		id = jsonDados['id'];
		idPessoa = jsonDados['idPessoa'];
		idCargo = jsonDados['idCargo'];
		idSetor = jsonDados['idSetor'];
		matricula = jsonDados['matricula'];
		dataCadastro = jsonDados['dataCadastro'] != null ? DateTime.tryParse(jsonDados['dataCadastro']) : null;
		dataAdmissao = jsonDados['dataAdmissao'] != null ? DateTime.tryParse(jsonDados['dataAdmissao']) : null;
		dataDemissao = jsonDados['dataDemissao'] != null ? DateTime.tryParse(jsonDados['dataDemissao']) : null;
		ctpsNumero = jsonDados['ctpsNumero'];
		ctpsSerie = jsonDados['ctpsSerie'];
		ctpsDataExpedicao = jsonDados['ctpsDataExpedicao'] != null ? DateTime.tryParse(jsonDados['ctpsDataExpedicao']) : null;
		ctpsUf = jsonDados['ctpsUf'] == '' ? null : jsonDados['ctpsUf'];
		observacao = jsonDados['observacao'];
		usuario = jsonDados['usuario'] == null ? null : new Usuario.fromJson(jsonDados['usuario']);
		pessoa = jsonDados['pessoa'] == null ? null : new Pessoa.fromJson(jsonDados['pessoa']);
		cargo = jsonDados['cargo'] == null ? null : new Cargo.fromJson(jsonDados['cargo']);
		setor = jsonDados['setor'] == null ? null : new Setor.fromJson(jsonDados['setor']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonDados = new Map<String, dynamic>();

		jsonDados['id'] = this.id ?? 0;
		jsonDados['idPessoa'] = this.idPessoa ?? 0;
		jsonDados['idCargo'] = this.idCargo ?? 0;
		jsonDados['idSetor'] = this.idSetor ?? 0;
		jsonDados['matricula'] = this.matricula;
		jsonDados['dataCadastro'] = this.dataCadastro != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.dataCadastro) : null;
		jsonDados['dataAdmissao'] = this.dataAdmissao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.dataAdmissao) : null;
		jsonDados['dataDemissao'] = this.dataDemissao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.dataDemissao) : null;
		jsonDados['ctpsNumero'] = this.ctpsNumero;
		jsonDados['ctpsSerie'] = this.ctpsSerie;
		jsonDados['ctpsDataExpedicao'] = this.ctpsDataExpedicao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.ctpsDataExpedicao) : null;
		jsonDados['ctpsUf'] = this.ctpsUf;
		jsonDados['observacao'] = this.observacao;
		jsonDados['usuario'] = this.usuario == null ? null : this.usuario.toJson;
		jsonDados['pessoa'] = this.pessoa == null ? null : this.pessoa.toJson;
		jsonDados['cargo'] = this.cargo == null ? null : this.cargo.toJson;
		jsonDados['setor'] = this.setor == null ? null : this.setor.toJson;
	
		return jsonDados;
	}
	

	String objetoEncodeJson(Colaborador objeto) {
	  final jsonDados = objeto.toJson;
	  return json.encode(jsonDados);
	}
	
}